Client/mods/deathmatch/logic/CClientModel.cpp

```diff
@@ -53,20 +63,45 @@ bool CClientModel::Deallocate(void)
     // ModelInfo must be valid
     if (pModelInfo->IsValid())
     {
-        if (m_eModelType == CCLIENTMODELPED)
+        if (m_eModelType == eClientModelType::PED)
         {
             // If some ped is using this ID, change him to CJ
             CClientPedManager*                       pPedManager = g_pClientGame->GetManager()->GetPedManager();
-            std::vector<CClientPed*>::const_iterator iter = pPedManager->IterBegin();
-            for (; iter != pPedManager->IterEnd(); iter++)
+            for (auto iter = pPedManager->IterBegin(); iter != pPedManager->IterEnd(); iter++)
+            {
+                CClientPed* pPed = *iter;
+                if (pPed->GetModel() == m_iModelID)
+                {
+                    if (pPed->IsStreamedIn())
+                    {
+                        pPed->StreamOutForABit();
+                    }
+                    pPed->SetModel(0);
+
+                    CLuaArguments Arguments;
+                    Arguments.PushNumber(m_iModelID);
+                    Arguments.PushNumber(0);
+                    pPed->CallEvent("onClientElementModelChange", Arguments, true);
+                }
+            }
+        }
+        else if (m_eModelType == eClientModelType::OBJECT)
+        {
+            CClientObjectManager* pObjectManager = g_pClientGame->GetManager()->GetObjectManager();
+            for (auto* pObject : pObjectManager->GetObjects())
             {
-                if ((*iter)->GetModel() == m_iModelID)
+                if (pObject->GetModel() == m_iModelID)
                 {
-                    if ((*iter)->IsStreamedIn())
+                    if (pObject->IsStreamedIn())
                     {
-                        (*iter)->StreamOutForABit();
+                        pObject->StreamOutForABit();
                     }
-                    (*iter)->SetModel(0);
+                    pObject->SetModel(1337);
+
+                    CLuaArguments Arguments;
+                    Arguments.PushNumber(m_iModelID);
+                    Arguments.PushNumber(1337);
+                    pObject->CallEvent("onClientElementModelChange", Arguments, true);
                 }
```

Comment:
Create a function for this, please.

Reply:
This part of code use lambda function in #1701